packages/shopify-codemod/transforms/arguments-to-args-spread.js

```diff
@@ -0,0 +1,65 @@
+export default function argumentsToArgsSpread({source}, {jscodeshift: j}, {printOptions = {quote: 'single'}}) {
+  function isArglessIIFE(path) {
+    return j.match(path, {
+      type: j.CallExpression.name,
+      callee: {
+        type: j.ArrowFunctionExpression.name,
+        params: (params) => params.length === 0,
+      },
+      arguments: (args) => args.length === 0,
+    });
+  }
+
+  function findNonIIFEScope(path) {
+    let scope = path.scope;
+    while (isArglessIIFE(scope.path.parentPath)) {
+      scope = scope.parent;
+    }
+
+    return scope.path;
+  }
+
+  function isFunctionScopeWithNoParams({scope}) {
+    return scope.node.params.length === 0;
+  }
+
+  function isFunctionScopeWithArgsSpread({node: {params}}) {
+    return (
+      j.RestElement.check(params[0]) &&
+      params[0].argument.name === 'args'
+    );
+  }
+
+  function hasSpreadableRootScope(path) {
+    const rootScopePath = findNonIIFEScope(path);
+    return isFunctionScopeWithNoParams(rootScopePath) || isFunctionScopeWithArgsSpread(rootScopePath);
+  }
+
+  function hasNoArgsVar(path) {
+    return path.scope.lookup('args') == null;
+  }
+
+  const sourceAST = j(source);
+
+  const argumentsReferences = sourceAST
+    .find(j.Identifier, {name: 'arguments'})
+    .filter(hasSpreadableRootScope)
```

Comment:
It would look like: `.filter(path => hasNoArgsVar(path) || hasSpreadableRootScope(path))`

I find it harder to read and most of the time function chaining is preferred (easier to read, easier to add and remove from the chain).